design_notes/XDN05 - xlang Binary Interface.md

```diff
@@ -12,6 +12,233 @@ status: draft
 
 ## Abstract
 
-This design note describes xlang's [application binary interface](https://en.wikipedia.org/wiki/Application_binary_interface)
+This design note describes xlang's
+[application binary interface](https://en.wikipedia.org/wiki/Application_binary_interface)
 (or ABI). The ABI defines a stable contract between xlang language projections and xlang components
 that is necessary in order to interoperate.
+
+## Overview
+
+xlang components are intended to be authored and built independently from their clients - both
+applications as well as other xlang components. Because an xlang component may be built at a
+different time and potentially using a different compiler from the calling code, xlang needs a
+stable contract - known as an application binary interface or ABI - to enable interoperability.
+
+xlang uses COM interfaces as the stable contract for xlang object types. Flat C functions are also
+used in xlang component binary loading as well as in xlang's
```

Comment:
Remove line feed.

Reply:
As the GitHub PR/diff viewer doesn't provide a more readable way to view the markdown, being a bit more careful about the source may help reviews.

Reply:
I tried the unified view for markdown and its more readable - thanks.